### PR TITLE
Rails api backend

### PR DESCRIPTION
**Pull request by mwitte98**
Rails api backend with CORS set up

### CODE REVIEW

File: client/.gitignore [hunk 1]
```diff
@@ -1,2 +0,0 @@
-node_modules
```
Comment (7imbrook) [inline client/.gitignore:1]: Bring this back pls

**Review by 7imbrook — CHANGES_REQUESTED** [review]
How much of this is really needed? Do we need a mailer template? Try to get this PR down in size :)

File: client/.gitignore (hunk 1 above)
Comment (mwitte98) [reply client/.gitignore:1]: I moved the .gitignore to the root and added node_modules there.

The previous team used mailers to let students know they've been added to a course. We'll probably also need mailers for forgot password. I can take a look through it again but the only thing I can think of removing is action cable if we aren't going to be using web sockets for anything.

Comment (7imbrook) [reply client/.gitignore:1]: Ok, I missed that sorry. I'd still like to remove the mailers until it becomes a feature. Avoid having unused coded.

**Comment by mwitte98** [comment]
Action cable, action mailer, active job, action view, i18n, sprockets, and empty folders removed. Ready to be reviewed.

**Review by 7imbrook — APPROVED** [review]

**Pull request merged**